tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/TypeTestExprTest.java

```diff
@@ -125,9 +125,18 @@ public void testTypeTestExprNegative() {
                 "unnecessary condition: expression will always evaluate to 'true'", 246, 9);
         BAssertUtil.validateError(negativeResult, i++,
                 "incompatible types: 'foo|bar' will not be matched to 'baz|2'", 255, 9);
-        BAssertUtil.validateError(negativeResult, i,
+        BAssertUtil.validateError(negativeResult, i++,
                 "incompatible types: '(string|int)' will not be matched to '(float|boolean)'",
                 262, 9);
+        BAssertUtil.validateError(negativeResult, i++,
```

Comment:
Given that the following is allowed

```ballerina
    xml<never> x = xml ``;
    'xml:Text t = x;
    string s1 = t;
```

shouldn't the following also be allowed?

```ballerina
    xml<never> x = xml ``;
    string s2 = x; 
```

Reply:
Since we don't allow:
```
xml x1 = xml `I am a string`;
string a = x1;
```
the 2nd one should not be allowed right?

Reply:
The spec doesn't talk about sub types of xml:Text being implicitly castable to string right?

Reply:
Whatever works for `T` should generally work for subtypes of `T` too. 

So since `xml<never>` is a subtype of `xml:Text` and `xml:Text` can be implicitly converted to `string`, `xml<never>` should be implicitly converted to `string`.

The [spec](https://ballerina.io/spec/lang/master/#section_6.3) also explicitly mentions the empty xml value under "Casting and conversion" 

> If a value of type xml:Text is the empty xml value, then it is implicitly converted to the empty string; otherwise the value is implicitly converted to a string that has a constituent character for each singleton item in the xml:Text sequence.